core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts

```diff
@@ -131,6 +131,49 @@ export function parseFromDotObjectToGridStack(
     })) as DotGridStackWidget[];
 }
 
+export const parseFromGridStackToDotObject = (gridData: DotGridStackWidget[]): DotLayoutBody => {
+    // Clone the data so we don't mutate the original
+    const clone = structuredClone(gridData);
+    const ordered = clone.sort((a, b) => a.y - b.y);
+
+    const rows = ordered.map((row) => {
+        const { x: colWidth, subGridOpts, styleClass: styles, containers } = row;
+        const { children = [] } = subGridOpts || {};
+        const styleClass = styles?.join(' ') || null;
+
+        // If there is no subgrid, we return a single column
+        const column = [
+            {
+                containers,
+                leftOffset: colWidth,
+                width: colWidth,
+                styleClass
+            }
+        ];
```

Comment:
Here is an small error, indeed row has the `containers` property, because we needed to add it in the custom type we are using, but that property will always come as undefined, because in the business logic, rows cannot contain containers. I think we should document that. 

Also, I'm not sure if we should fill the row with a column that has empty containers. @fmontes what do you think?

Reply:
Ok, done!